QModManager/API/SMLHelper/Patchers/OptionsPanelPatcher.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace QModManager.API.SMLHelper.Patchers
+{
+    using Harmony;
+    using Options;
+    using System.Collections.Generic;
+    using UnityEngine;
+    using UnityEngine.Events;
+    using Logger = QModManager.Utility.Logger;
+
+    internal static class OptionsPanelPatcher
+    {
+        internal static SortedList<string, ModOptions> modOptions = new SortedList<string, ModOptions>();
+
+        internal static void Patch(HarmonyInstance harmony)
+        {
+            harmony.Patch(AccessTools.Method(typeof(uGUI_OptionsPanel), "AddTabs"),
+                postfix: new HarmonyMethod(AccessTools.Method(typeof(OptionsPanelPatcher), "AddTabs_Postfix")));
+
+            Logger.Debug("OptionsPanelPatcher is done.");
+        }
+
+        internal static void AddTabs_Postfix(uGUI_OptionsPanel __instance)
+        {
+            uGUI_OptionsPanel optionsPanel = __instance;
+
+            // Start the modsTab index at a value of -1
+            var modsTab = OptionsManager.ModsTab;
+
+            optionsPanel.AddChoiceOption(modsTab, "Extra item info", new string[]
+            {
+                "Mod name (default)",
+                "Mod name and item ID",
+                "Nothing"
+            }, (int)TooltipPatcher.ExtraItemInfoOption, (i) => TooltipPatcher.SetExtraItemInfo((TooltipPatcher.ExtraItemInfo)i));
+
+            foreach (ModOptions modOption in modOptions.Values)
+            {
+                optionsPanel.AddHeading(modsTab, modOption.Name);
+
+                foreach (ModOption option in modOption.Options)
+                {
+                    switch (option.Type)
+                    {
+                        case ModOptionType.Slider:
+                            var slider = (ModSliderOption)option;
+
+                            optionsPanel.AddSliderOption(modsTab, slider.Label, slider.Value, slider.MinValue, slider.MaxValue, slider.Value,
+                                new UnityAction<float>((float sliderVal) =>
+                                    modOption.OnSliderChange(slider.Id, sliderVal)));
+                            break;
```

Comment:
Bit complicated, maybe split into methods?

Reply:
That _is_ a method call.  
If this is to be made simpler and easier to reach, we should consider combining all those parameters into a single class that can be passed directly.

Reply:
No, that is a patch, not a method call. By method calls i mean separate methods for each type of option